modules/headless_ai/cfgFunctions/create/fn_createEmptyVehicle.sqf

```diff
@@ -1,57 +1,59 @@
 #include "..\..\script_macros.hpp"
 
-
 params [
-    "_vehClass",
-    "_vehPos",
-    "_vectorDir",
-    "_vectorUp",
-    "_damage",
-    "_fuel",
-    "_turretMags",
-    "_locked",
-    "_vehInWater",
-    "_vehName",
-    "_persistent",
-    "_vehInit",
-    "_storedVars",
+    ["_vehClass", "", [""]],
+    ["_vehCrew", [], [[]]],
+    ["_pos", [], [[]]],
+    ["_vectorDir", [], [[]]],
+    ["_vectorUp", [], [[]]],
+    ["_damage", 0, [0]],
+    ["_fuel", 100, [100]],
+    ["_turretMags", [], [[]]],
+    ["_locked", 0, [0]],
+    ["_init", false, [false,{}]],
+    ["_fly", false, [false]],
+    ["_flyInHeight", 200, [200]],
+    ["_storedVars", [], [[]]],
     ["_vehCustomization", [], [[]]],
-    "_name",
+    ["_varName", "", [""]],
     ["_olsenGearType", "", [""]]
 ];
 
-private _vehicle = createVehicle [_vehClass,_vehPos,[],0,"CAN_COLLIDE"];
+private _vehicle = createVehicle [_vehClass,_pos,[],0,"CAN_COLLIDE"];
 _vehicle setVectorDirAndUp [_vectorDir, _vectorUp];
-_vehicle setPosATL _vehPos;
+_vehicle setPosATL _pos;
 
-if !(_name isEqualTo "") then {
-    private _uniqueName = [_name] call FUNC(findUniqueName);
+if (_varName isNotEqualTo "") then {
+    private _uniqueName = [_varName] call FUNC(findUniqueName);
     missionNamespace setVariable [_uniqueName, _vehicle, true];
 };
 
-if !(_olsenGearType isEqualTo "") then {
-    [_vehicle, _olsenGearType] call EFUNC(FW,VehGearScript);
+if (_olsenGearType isNotEqualTo "") then {
+    [_vehicle, _olsenGearType] call FUNC(VehicleGearScript);
 };
 
 _vehicle setDamage _damage;
 _vehicle setFuel _fuel;
-_turretMags apply {
-    _x params [["_class",
-"",[""]],["_path",[],[[]]],["_ammo",0,[0]]];
+{
+    _x params [["_class","",[""]],["_path",[],[[]]],["_ammo",0,[0]]];
     _vehicle setMagazineTurretAmmo [_class,_ammo,_path];
-};
+} forEach _turretMags;
```

Comment:
Why did you revert nice things like that?

Reply:
port of some mmfw functions

Reply:
You know better than me, is there a noticeable performance impact by leaving it as forEach or should we just wait for next update to gradually convert?

Reply:
already changed